Issue 1284 - Drop AWS SDKv1

barryoneill opened the pull request.

barryoneill committed:
Issue #1284 - drop AWS SDKv1 support

# Conflicts:
#	fs2-aws-dynamodb/src/main/scala/fs2/aws/dynamodb/CommittableRecord.scala
#	fs2-aws-dynamodb/src/main/scala/fs2/aws/dynamodb/DynamoDB.scala

barryoneill committed:
wip

barryoneill committed:
fix compile

barryoneill committed:
fix algebra generation for new sdk

# Conflicts:
#	build.sbt
#	project/Dependencies.scala

tmilner commented:
I was just about to do this, so this is great timing. Seems like you have a lot of unused imports causing build failures tho :)

barryoneill committed:
Work on tagless gen imports

barryoneill committed:
Explicitly list out return types to suppress warnings

barryoneill committed:
Fix test by waiting longer

NOT a bug The KCL itself is much slower to init :/  Looks like the `CLIENT_VERSION_CONFIG_COMPATIBLE_WITH` migration logic takes time, the secondary GSI creation adds time, and the lease assignment loop takes longer :/

barryoneill committed:
fix formatting

barryoneill committed:
doc fixes

matejcerny commented:
You probably all know this, but the KPL reached end-of-support on January 30, 2026
